### PR TITLE
Spelling updates before 0.0.23

### PR DESCRIPTION
**Pull request by jsoref**
### Short description

I'm about to release 0.0.23, these changes were developed using it (see https://github.com/check-spelling-sandbox/pdns/actions/workflows/spelling3.yml?query=branch%3Acheck-spelling-0.0.23)

* Roughly merging configuration from https://github.com/check-spelling/spell-check-this/commit/f398e2297a86886bcb3f04e6627ea57d969a80b1
* Adding `regression-tests.*/description` to only to enable checking (based on a comment in `excludes.txt` which I used to move to `only.txt`)
* Turning on `check_file_names` to check file name paths (this is why there a bunch of file renames in this PR)
* Adding a `lib*` pattern (which reduces the expect file by a bunch of tokens)
* Adding a `C network byte conversions` pattern (which removes the `hto*` tokens)

None of the changes above require the newer version of check-spelling (but upgrading to it, the PR for that should come ~ Wednesday) should cut ~90s from the run time

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/recursordist/docs/lua-config/proxymapping.rst [hunk 1]
```diff
@@ -48,7 +48,7 @@ See :func:`protobufServer` on how to tune the source address logged in ``Protobu
   Specify a table based mapping for a subnet.
 
   :param string subnet: a subnet to match
-  :param string ip: the IP address or IPaddress port combination to match the subnet to.
+  :param string ip: the IP address or IP address + port combination to match the subnet to.
```
Comment (jsoref) [inline pdns/recursordist/docs/lua-config/proxymapping.rst:51]: This was masked by the previous rule... The choice of `+` was arbitrary -- I'm definitely not attached to it.

File: regression-tests/tests/1dyndns-content-mixed-case/command
Comment (jsoref) [inline regression-tests/tests/1dyndns-content-mixed-case/command:1]:
```diff
-regression-tests/tests/1dyndns-content-casemix/command
+regression-tests/tests/1dyndns-content-mixed-case/command
```

File: regression-tests/tests/1dyndns-update-add-delete-case-sensitive/command
Comment (jsoref) [inline regression-tests/tests/1dyndns-update-add-delete-case-sensitive/command:1]:
```diff
-regression-tests/tests/1dyndns-update-add-delete-casesensative/command
+regression-tests/tests/1dyndns-update-add-delete-case-sensitive/command
```

File: regression-tests/tests/mx-case-sensitivity-with-ap/command
Comment (jsoref) [inline regression-tests/tests/mx-case-sensitivity-with-ap/command:1]:
```diff
-regression-tests/tests/mx-case-sensitivy-with-ap/command
+regression-tests/tests/mx-case-sensitivity-with-ap/command
```

File: .github/actions/spell-check/only.txt [hunk 2]
```diff
@@ -2,3 +2,4 @@
 ^docs/
 \.md$
 pdns/recursordist/settings/table.py
+regression-tests.*/description
```
Comment (jsoref) [inline .github/actions/spell-check/only.txt:5]: As noted, this was moved from `excludes.txt` and was a valuable move -- also, this change in concert with the check-file-names feature is what caught the directory changes below...

**Review by Habbie — APPROVED** [review]

**Pull request merged**